src/java/com/sarality/app/datastore/contact/ContactDataStore.java

```diff
@@ -203,8 +166,50 @@ private ContactDataBuilder extractContactData(ContactDataBuilder builder, long c
       nameBuilder.setMiddleName(processors.forString().extract(cursor, Column.DATA5));
       nameBuilder.setSuffix(processors.forString().extract(cursor, Column.DATA6));
       builder.setName(nameBuilder);
+    } else {
+      for (AppEnum app : AppEnum.values()) {
```

Comment:
This works for now because you have only 1 entry but what you might want to consider is a way to lookup AppEnum using mimeType. If that value is not null add an App Contact.

At least add a TODO to do that